packages/@uppy/url/src/Url.jsx

```diff
@@ -48,7 +48,7 @@ function checkIfCorrectURL (url) {
 }
 
 function getFileNameFromUrl (url) {
-  return url.substring(url.lastIndexOf('/') + 1)
+  return url.substring(url.lastIndexOf('/') + 1).split('?')[0], // make sure filename does not have query params in it
```

Comment:
That won't work as expected if there's a `/` after the `?` char. Instead we can do
```suggestion
  const queryParamIndexOf = url.indexOf('?')
  const queryParamStart = queryParamIndexOf === -1 ? undefined : queryParamIndexOf
  return url.substring(url.lastIndexOf('/', queryParamStart) + 1, queryParamStart)
```

Reply:
> Wouldn't calling [uppy.setFileMeta](https://uppy.io/docs/uppy/#uppy-setFileMeta-fileID-data) work to add meta?

I don't think so.  The reason is sequencing or WHEN.  As I mentioned in the PR comments,  there's no intermediate opportunity to add data before uppy.addFile gets called.  When upp.addFile gets called, there are registered listeners that receive the File object BEFORE I have a chance to add meta data to it.